test/integration/crud/bulk.test.ts

```diff
@@ -1688,6 +1688,55 @@ describe('Bulk', function () {
     }
   });
 
+  it('should apply hint via FindOperators', {
+    metadata: { requires: { mongodb: '>= 4.2' } },
+    async test() {
+      const bulk = client.db().collection('coll').initializeOrderedBulkOp();
+
+      const events = [];
+      client.on('commandStarted', event => {
+        if (['update', 'delete'].includes(event.commandName)) {
+          events.push(event);
+        }
+      });
+
+      // updates
+      bulk
+        .find({ b: 1 })
+        .hint({ b: 1 })
+        .updateOne({ $set: { b: 2 } });
+      bulk
+        .find({ b: 2 })
+        .hint({ b: 1 })
+        .update({ $set: { b: 3 } });
+      bulk.find({ b: 3 }).hint({ b: 1 }).replaceOne({ b: 2 });
+
+      // deletes
+      bulk.find({ b: 2 }).hint({ b: 1 }).deleteOne();
+      bulk.find({ b: 1 }).hint({ b: 1 }).delete();
+
+      await bulk.execute();
+
+      try {
+        expect(events).to.be.an('array').with.length.at.least(1);
+        expect(events[0]).property('commandName').to.equal('update');
+        const updateCommand = events[0].command;
+        expect(updateCommand).property('updates').to.be.an('array').with.length(3);
+        updateCommand.updates.forEach(statement => {
+          expect(statement).property('hint').to.eql({ b: 1 });
+        });
+        expect(events[1]).property('commandName').to.equal('delete');
+        const deleteCommand = events[1].command;
+        expect(deleteCommand).property('deletes').to.be.an('array').with.length(2);
+        deleteCommand.deletes.forEach(statement => {
+          expect(statement).property('hint').to.eql({ b: 1 });
+        });
+      } finally {
+        await client.close();
+      }
```

Comment:
There's an afterEach hook that handles closing the client for each of these tests

Reply:
Done, and ditto for the other added test.

Nearly every test in this file explicitly closes the client, though. Happy to do a follow-up PR where this is aligned across the file.

Reply:
This file used to explicitly connect at the top of each test so that's why there's a close to mirror it, we cleaned up the connects but not closes, no harm in closing more than once but we don't need to in new tests. 

That'd be nice :) but probably not worth the time